go/types/validating_batching_sink.go

```diff
@@ -16,28 +19,65 @@ type ValidatingBatchingSink struct {
 	cs    chunks.ChunkStore
 	batch [batchSize]chunks.Chunk
 	count int
-	tc    *TypeCache
+	pool  sync.Pool
 }
 
-func NewValidatingBatchingSink(cs chunks.ChunkStore, tc *TypeCache) *ValidatingBatchingSink {
-	return &ValidatingBatchingSink{vs: newLocalValueStore(cs), cs: cs, tc: tc}
+func NewValidatingBatchingSink(cs chunks.ChunkStore) *ValidatingBatchingSink {
+	return &ValidatingBatchingSink{
+		vs:   newLocalValueStore(cs),
+		cs:   cs,
+		pool: sync.Pool{New: func() interface{} { return NewTypeCache() }},
```

Comment:
nevermind. i get it. the validatingbatchingsink only lives the lifetime of the handleWrite and then all of the typeCaches get dropped, right?